<h1 style="color:navy; text-align:center;">Uncovering Cancer Trends in England: A Decade of NHS Insights</h1>


## 📖 Introduction

Cancer remains one of the most significant health challenges in England, affecting thousands of individuals each year. The NHS collects comprehensive data on cancer diagnoses, which provides valuable insights into patterns and trends across the population. This project analyzes NHS cancer statistics from 2013 to 2022, with a focus on factors such as cancer type, stage at diagnosis, gender, and the distribution of cases among different population groups. By examining these patterns, the project aims to understand how cancer incidence varies across the population and how it has evolved over the past decade. While the primary goal is to gain a deeper understanding of cancer trends, the insights generated could also help inform NHS strategies related to early detection, prevention, and improving patient outcomes. Through this analysis, the project seeks to uncover important trends, highlight areas of concern, and provide a data-driven perspective on how cancer impacts various segments of the population.

## 🎯 Aim 

To analyze NHS cancer statistics from 2013 to 2022 in order to understand the distribution and trends of cancer incidence across different population groups, considering factors such as cancer type, stage at diagnosis, and gender, and to generate insights that could support a better understanding of how cancer affects the population.

## 🏥 Objective

1. To explore and describe trends in cancer incidence over the past decade.

2. To examine the distribution of different cancer types among various population groups.

3. To analyze differences in cancer incidence between genders.

4. To identify population groups that may be disproportionately affected by cancer.

5. To provide visualizations and data-driven insights that help understand the overall cancer landscape in England.

## 💡Data Snapshot: Overview of the Data

In [1]:
import pandas as pd

In [3]:
df = pd.read_csv(r"C:\Users\Lenovo\OneDrive\Desktop\Project\NHS_CancerData.csv")

C:\Users\Lenovo\AppData\Local\Temp\ipykernel_13596\3578574885.py:1: DtypeWarning: Columns (8,9) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(r"C:\Users\Lenovo\OneDrive\Desktop\Project\NHS_CancerData.csv")


In [4]:
df.head()

diagnosisyear geography_type geography_code geography_name ndrs_main_group  \
0           2013        Country      E92000001        England            Anus   
1           2013        Country      E92000001        England            Anus   
2           2013        Country      E92000001        England            Anus   
3           2013        Country      E92000001        England            Anus   
4           2013        Country      E92000001        England            Anus   

  ndrs_detailed_group stage_at_diagnosis       imd_quintile hormone_receptor  \
0            All Anus         All stages      All quintiles              NaN   
1            All Anus         All stages      All quintiles              NaN   
2            All Anus         All stages      All quintiles              NaN   
3            All Anus         All stages  1 - most deprived              NaN   
4            All Anus         All stages                  2              NaN   

  hormone_receptor_status   gender age_at_diagnosis  count  Population  \
0                     NaN  Persons         All ages   1057    53918686   
1                     NaN  Females         All ages    668    27410034   
2                     NaN    Males         All ages    389    26508652   
3                     NaN  Persons         All ages    232    10887630   
4                     NaN  Persons         All ages    219    10952016   

       type_of_rate  rate  lower_confidence_interval  \
0  Age-standardised   2.1                        2.0   
1  Age-standardised   2.6                        2.4   
2  Age-standardised   1.7                        1.5   
3  Age-standardised   2.8                        2.5   
4  Age-standardised   2.5                        2.2   

   upper_confidence_interval flag  
0                        2.3  NaN  
1                        2.8  NaN  
2                        1.9  NaN  
3                        3.2  NaN  
4                        2.8  NaN

*✅ The first few rows show cancer diagnosis records, including details on year, geography, cancer type, stage, demographics, and statistical indicators such as counts and rates.*

In [5]:
df.shape

(35030, 19)

✅ *The dataset comprises 35,030 rows and 19 columns.*

In [7]:
df.columns

Index(['diagnosisyear', 'geography_type', 'geography_code', 'geography_name',
       'ndrs_main_group', 'ndrs_detailed_group', 'stage_at_diagnosis',
       'imd_quintile', 'hormone_receptor', 'hormone_receptor_status', 'gender',
       'age_at_diagnosis', 'count', 'Population', 'type_of_rate', 'rate',
       'lower_confidence_interval', 'upper_confidence_interval', 'flag'],
      dtype='object')

✅ *The dataset includes variables capturing diagnosis year, geography, cancer type, stage, demographics, incidence counts, population, and rate estimates.*

In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 35030 entries, 0 to 35029
Data columns (total 19 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   diagnosisyear              35030 non-null  int64  
 1   geography_type             35030 non-null  object 
 2   geography_code             35030 non-null  object 
 3   geography_name             35030 non-null  object 
 4   ndrs_main_group            35030 non-null  object 
 5   ndrs_detailed_group        35030 non-null  object 
 6   stage_at_diagnosis         35030 non-null  object 
 7   imd_quintile               35030 non-null  object 
 8   hormone_receptor           690 non-null    object 
 9   hormone_receptor_status    690 non-null    object 
 10  gender                     35030 non-null  object 
 11  age_at_diagnosis           35030 non-null  object 
 12  count                      35030 non-null  int64  
 13  Population                 35030 non-null  int

✅ *The dataset contains 35,030 rows and 19 columns. Out of these, 13 are categorical, 3 are integers (diagnosisyear, count, Population), and 3 are decimals (rate, lower_confidence_interval, upper_confidence_interval).*

*Some fields are quite sparse:*

*1)The hormone receptor fields are missing for almost the entire dataset (around 98% missing).*

*2)The flag column is missing in nearly 80% of rows.*

*The age_at_diagnosis column is stored as text, suggesting it may represent age bands rather than exact ages.*


In [9]:
df.describe()

diagnosisyear          count    Population          rate  \
count   35030.000000   35030.000000  3.503000e+04  29792.000000   
mean     2017.500000    1497.867713  2.124806e+07     12.719364   
std         2.872322    6697.796599  1.757073e+07     43.901143   
min      2013.000000       0.000000  5.201813e+06      0.000000   
25%      2015.000000      32.000000  5.650682e+06      0.500000   
50%      2017.500000     166.000000  1.137309e+07      1.900000   
75%      2020.000000     740.000000  2.830780e+07      7.700000   
max      2022.000000  230968.000000  5.711254e+07    673.600000   

       lower_confidence_interval  upper_confidence_interval  
count               29792.000000               29792.000000  
mean                   12.260382                  13.210795  
std                    43.332517                  44.474403  
min                     0.000000                   0.000000  
25%                     0.400000                   0.700000  
50%                     1.700000                   2.200000  
75%                     7.100000                   8.300000  
max                   666.700000                 680.500000

✅*The dataset covers cancer diagnoses from 2013 to 2022.*

*Case counts range from 0 to 230,968, with most records showing relatively small numbers.*

*Population figures span from about 5 million to 57 million, with a middle value around 11 million.*

*The rate column shows how common cancer is in the population (cases per 100,000). Most rates are low, with a median of about 1.9, but a few records have much higher rates, reaching up to 673.6.*

*The confidence interval columns show the likely range for the rate. Most values fall between 0.4 and 8.3, though a few extreme records have much higher ranges, above 600.*